Google.Solutions.Compute/Auth/Authorization.cs

```diff
@@ -0,0 +1,184 @@
+﻿//
+// Copyright 2019 Google LLC
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Flows;
+using Google.Apis.Util.Store;
+using System;
+using System.Threading.Tasks;
+using System.Threading;
+using System.IO;
+
+namespace Google.Solutions.Compute.Auth
+{
+    public interface IAuthorization
+    {
+        ICredential Credential { get; }
+
+        string UserId { get; }
+
+        Task RevokeAsync();
+    }
+
+    public class OAuthAuthorization : IAuthorization
+    {
+        public const string StoreUserId = "oauth";
+
+        private readonly UserCredential credential;
+        private readonly IDataStore credentialStore;
+
+        public OAuthAuthorization(
+            UserCredential credential,
+            IDataStore credentialStore)
+        {
+            this.credential = credential;
+            this.credentialStore = credentialStore;
+        }
+
+        public ICredential Credential => this.credential;
+
+        public string UserId => this.credential.UserId;
+
+        public Task RevokeAsync()
```

Comment:
we're deleting the creds on revoke...do we also delete the credentials from the registry if the plugin is 'uninstalled'

Reply:
Hmm.. that's difficult because whoever runs the uninstall might not have access to all user's HKCU hives. I think it would also run counter to the idea of having the credentials part of the roaming profile... if I uninstall the plugin on one machine, I might still want to use it on a different machine (using the same roaming user profile).